stl/inc/compare

```diff
@@ -78,38 +64,36 @@ public:
     }
 
     _NODISCARD friend constexpr bool operator<=(const partial_ordering _Val, _Literal_zero) noexcept {
-        return _Val._Value <= 0 && _Val._Is_ordered();
+        return static_cast<signed char>(static_cast<unsigned char>(_Val._Value) - 1) < 0;
     }
 
     _NODISCARD friend constexpr bool operator>=(const partial_ordering _Val, _Literal_zero) noexcept {
         return _Val._Value >= 0;
     }
 
     _NODISCARD friend constexpr bool operator<(_Literal_zero, const partial_ordering _Val) noexcept {
-        return 0 < _Val._Value;
+        return _Val._Value > 0;
     }
 
     _NODISCARD friend constexpr bool operator>(_Literal_zero, const partial_ordering _Val) noexcept {
-        return 0 > _Val._Value && _Val._Is_ordered();
+        return _Val._Value == static_cast<_Compare_t>(_Compare_ord::less);
     }
 
     _NODISCARD friend constexpr bool operator<=(_Literal_zero, const partial_ordering _Val) noexcept {
-        return 0 <= _Val._Value;
+        return _Val._Value >= 0;
     }
 
     _NODISCARD friend constexpr bool operator>=(_Literal_zero, const partial_ordering _Val) noexcept {
-        return 0 >= _Val._Value && _Val._Is_ordered();
+        return static_cast<signed char>(static_cast<unsigned char>(_Val._Value) - 1) < 0;
     }
 
-#ifdef __cpp_impl_three_way_comparison
     _NODISCARD friend constexpr partial_ordering operator<=>(const partial_ordering _Val, _Literal_zero) noexcept {
         return _Val;
     }
 
     _NODISCARD friend constexpr partial_ordering operator<=>(_Literal_zero, const partial_ordering _Val) noexcept {
-        return partial_ordering{static_cast<_Compare_ord>(-_Val._Value)};
+        return partial_ordering{static_cast<_Compare_ord>(0 - static_cast<unsigned>(_Val._Value))};

```

Comment:
This is also very clever (I had to work out what it does to all of the input values to see how it exchanges `less` and `greater` while leaving `equivalent` and `unordered` unchanged). :smile_cat:

Reply:
Yes, it took me a minute or two to figure out how to negate `CHAR_MIN` into `CHAR_MIN` without undefined behavior.